docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1459 @@
+  config: |
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "mtu": 1500,
+            "netAttachDefName": "default/l3-network",
+            "role": "primary",
+            "transport": "noOverlay"
+    }
+```
+
+##### Host Routing Tables
+
+To [maintain isolation between
+UDNs](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5140), node BGP
+speakers do not accept routes to pod subnets by default. However, to route
+east-west traffic in no-overlay mode, these routes must be imported to the host
+routing table.
+
+##### Gateway Router
+
+Currently, BGP routes to other pod subnets are not imported into the gateway
+router, which ensures node-to-pod traffic traverses the OVN logical topology.
+However, we need the routes to be imported into the gateway router in no-overlay
+mode. We need to update the route import controller to check whether a network
+is in no-overlay mode and import the routes from the host routing table if so.
+
+#### Configurable SNAT for Egress Traffic
+
+The OVN-Kubernetes BGP feature currently disables SNAT for all pod egress
+traffic if the pod network is advertised. The only exception is for traffic
+destined to other nodes in the cluster (pod-to-other-node), to ensure nodePort
+services can be accessed across networks. This egress traffic will remain SNATed
+to the node IP before leaving the node.
+
+However, this approach assumes that all external destinations can route back to
+the pod IPs, which is often not true for destinations reached via non-BGP
+routers (like a default gateway). To give users control over pod egress traffic
+SNAT behavior, we introduce a new `outboundSNAT` field to the
+ClusterUserDefinedNetwork CRD. When enabled, pod egress traffic will be SNATed
+to the node IP before leaving the node.
+
+The `outboundSNAT` field is intended for deployments where a user wants a
+no-overlay network but has not advertised the pod network to be routable from
+external networks. When enabled, pod-to-external traffic is SNATed and forwarded
+by a default gateway, while pod-to-remote-pod traffic is not SNATed regardless
+of the value of this field.
+
+##### Shared Gateway Mode
+
+If `outboundSNAT` is enabled for a network, in the gateway router, the default
+SNAT rule will be kept for pod-to-external traffic. But pod-to-remote-pod
+traffic remains unaffected by SNAT. Therefore, the network controllers in
+ovnkube-controller will need to add an address set for the pod subnets of each
+CUDN and the default network. Then the network controllers will add an exemption
+rule with the address set to the SNAT entries. We use the `exempted_ext_ips`
+field instead of adding a negative match to the `match` field, as using a
+negative match in an OVN SNAT rule is not efficient. Setting the
+`exempted_ext_ips` field results in one OpenFlow rule being created for each pod
+SNAT entry in the OVS bridge.
```

Comment:
nit:

The phrase starts with `If `outboundSNAT` is enabled for a network` and that's important. If the network has geneve transport or `outboundSNAT` disabled, we shouldn't be doing any of this to avoid the risk of impacting scale. I am not sure if this could be conveyed more clearly. Same for local gateway mode where this is even less explicitly stated.